File: src/main/java/plugin/interaction/combat/magic/modern/teleport/TrollheimTeleportSpell.java
package plugin.interaction.combat.magic.modern.teleport;

import org.redrune.game.global.WorldTile;
import org.redrune.game.content.plugin.combat.spell.type.TeleportSpellPlugin;

/**
 * @author Tyluur <itstyluur@icloud.com>
 * @since 7/27/2017
 */
public class TrollheimTeleportSpell implements TeleportSpellPlugin {
	
	@Override
	public int levelRequired() {
		return 61;
	}
	
	@Override
	public int[] runesRequired() {
		return arguments(FIRE_RUNE, 2, LAW_RUNE, 2);
	}
	
	@Override
	public WorldTile destination() {
		return new WorldTile(2888, 3674, 0);
	}
	
	@Override
	public int spellId() {
		return 69;
	}
	
	@Override
	public double exp() {
		return 68;
	}
	
	@Override
	public MagicBook book() {
		return MagicBook.REGULAR;
	}
}

File: src/main/kotlin/plugin/interaction/combat/magic/modern/teleport/TrollheimTeleportSpell.kt
package plugin.interaction.combat.magic.modern.teleport

import org.redrune.game.content.plugin.combat.spell.type.TeleportSpellPlugin
import org.redrune.game.global.WorldTile
import org.redrune.utility.constants.MagicConstants
import org.redrune.utility.constants.MagicConstants.MagicBook

/**
 * @author Tyluur <itstyluur></itstyluur>@icloud.com>
 * @since 7/27/2017
 */
class TrollheimTeleportSpell : TeleportSpellPlugin {
    override fun levelRequired(): Int {
        return 61
    }

    override fun runesRequired(): IntArray {
        return arguments(MagicConstants.FIRE_RUNE, 2, MagicConstants.LAW_RUNE, 2)
    }

    override fun destination(): WorldTile {
        return WorldTile(2888, 3674, 0)
    }

    override fun spellId(): Int {
        return 69
    }

    override fun exp(): Double {
        return 68.0
    }

    override fun book(): MagicBook {
        return MagicBook.REGULAR
    }
}